test/integration/replicaset/test_replicaset_upgrade.py

```diff
@@ -0,0 +1,245 @@
+    return output
+
+
+@pytest.mark.skipif(
+    tarantool_major_version < 3, reason="skip centralized config test for Tarantool < 3"
+)
+def test_upgrade_cluster(tt_cmd, tmpdir_with_cfg):
+    tmpdir = tmpdir_with_cfg
+    app_name = "test_ccluster_app"
+    app_path = os.path.join(tmpdir, app_name)
+    shutil.copytree(os.path.join(os.path.dirname(__file__), app_name), app_path)
+
+    replicasets = [
+        "replicaset-001",
+        "replicaset-002",
+    ]
+
+    try:
+        # Start a cluster.
+        start_cmd = [tt_cmd, "start", app_name]
+        rc, out = run_command_and_get_output(start_cmd, cwd=tmpdir)
+        assert rc == 0
+
+        for i in range(1, 6):
+            file = wait_file(
+                os.path.join(tmpdir, app_name), f"ready-instance-00{i}", []
+            )
+            assert file != ""
+
+        _ = run_command_on_instance(
+            tt_cmd, tmpdir, "test_ccluster_app:instance-004", "box.cfg{read_only=true}"
+        )
+
+        # Instance bootstrap may not have finished yet.
+        sleep(5)
+        upgrade_cmd = [tt_cmd, "replicaset", "upgrade", app_name, "-t=10"]
+
+        rc, out = run_command_and_get_output(upgrade_cmd, cwd=tmpdir)
+        assert rc == 0
+
+        upgrade_out = out.strip().split("\n")
+        assert len(upgrade_out) == len(replicasets)
+
+        for i in range(len(replicasets)):
+            match = re.search(r"•\s*(.*?):\s*(.*)", upgrade_out[i])
+            assert match.group(1) in replicasets
+            assert match.group(2) == "ok"
+
+    finally:
+        stop_application(tt_cmd, app_name, tmpdir, [])
+
+
+@pytest.mark.skipif(
+    tarantool_major_version < 3, reason="skip centralized config test for Tarantool < 3"
+)
+def test_upgrade_multi_master(tt_cmd, tmpdir_with_cfg):
+    tmpdir = tmpdir_with_cfg
+    app_name = "test_ccluster_app"
+    app_path = os.path.join(tmpdir, app_name)
+    shutil.copytree(os.path.join(os.path.dirname(__file__), app_name), app_path)
+    try:
+        # Start a cluster.
+        start_cmd = [tt_cmd, "start", app_name]
+        rc, out = run_command_and_get_output(start_cmd, cwd=tmpdir)
+        assert rc == 0
+
+        for i in range(1, 6):
+            file = wait_file(
+                os.path.join(tmpdir, app_name), f"ready-instance-00{i}", []
+            )
+            assert file != ""
+
+        status_cmd = [tt_cmd, "replicaset", "upgrade", app_name]
+
+        rc, out = run_command_and_get_output(status_cmd, cwd=tmpdir)
+        assert rc == 1
+        assert "replicaset-002: error" in out and "are both masters" in out
+
+    finally:
+        stop_application(tt_cmd, app_name, tmpdir, [])
+
+
+@pytest.mark.skipif(
+    tarantool_major_version < 3, reason="skip centralized config test for Tarantool < 3"
+)
+def test_upgrade_t2_app_dummy_replicaset(tt_cmd):
+    app_name = "single-t2-app"
+    test_app_path_src = os.path.join(os.path.dirname(__file__), app_name)
+
+    # snapshot from tarantool 2.11.4 app
```

Comment:
Please, start all comments with a capital letter and end with a dot.
```suggestion
    # Snapshot from tarantool 2.11.4 app.
```

Reply:
Removed based on feedback in https://github.com/tarantool/tt/pull/936#discussion_r1818533222